core/src/KokkosExp_MDRangePolicy.hpp

```diff
@@ -62,6 +64,46 @@
 #include <ROCm/KokkosExp_ROCm_IterateTile_Refactor.hpp>
 #endif
 
+namespace Kokkos {
+namespace Impl {
+
+// Workaround for narrowing conversion warnings
+template <class IndexType>
+struct _ignore_narrowing_index_wrapper {
+  IndexType value;
+
+  template <class T,
+            int = typename std::enable_if<
+                std::is_convertible<T, IndexType>::value, int>::type{0}>
+  /* intentionally implicit */
+  KOKKOS_FORCEINLINE_FUNCTION _ignore_narrowing_index_wrapper(
+      T arg_value) noexcept
+      : value(arg_value) {}
+
+  // workaround for Cuda linker bug??!?
+  KOKKOS_INLINE_FUNCTION _ignore_narrowing_index_wrapper() noexcept
+      : value(IndexType{}) {}
+  KOKKOS_INLINE_FUNCTION _ignore_narrowing_index_wrapper(
+      _ignore_narrowing_index_wrapper const&) noexcept = default;
+  KOKKOS_INLINE_FUNCTION _ignore_narrowing_index_wrapper(
+      _ignore_narrowing_index_wrapper&&) noexcept = default;
+
+  KOKKOS_INLINE_FUNCTION _ignore_narrowing_index_wrapper& operator=(
+      _ignore_narrowing_index_wrapper const&) noexcept = default;
+  KOKKOS_INLINE_FUNCTION _ignore_narrowing_index_wrapper& operator=(
+      _ignore_narrowing_index_wrapper&&) noexcept = default;
+
+  KOKKOS_INLINE_FUNCTION ~_ignore_narrowing_index_wrapper() noexcept = default;
+
+  KOKKOS_INLINE_FUNCTION operator IndexType() noexcept { return value; }
+  KOKKOS_INLINE_FUNCTION operator IndexType const() const noexcept {
+    return value;
+  }
```

Comment:
Not sure how this used yet but why are these conversion operators not returning references?

Reply:
I could just remove this.  I'm hoping no one was using `point_type` directly, but just in case I thought I'd give them a conversion operator.